scripts/troubleshoot/cluster-migration/migrate-legacy-to-msi-aks.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+#
+# Script to migrate AKS clusters from service principal to managed identity authentication
+#
+# Usage:
+#   1. Create a text file listing your clusters (e.g., clusters.txt):
+#      ResourceGroup,ClusterName
+#      myRG-prod,cluster-prod
+#
+#   2. Run the script:
+#      ./migrate-legacy-to-msi-aks.sh clusters.txt
+#
+
+if [ "$1" == "--help" ] || [ "$1" == "-h" ] || [ -z "$1" ]; then
+    echo "Migrates AKS clusters to managed identity authentication"
+    echo ""
+    echo "Usage: $0 <clusters-file>"
+    echo "The clusters file should contain one cluster per line in format: resource-group,cluster-name"
+    echo ""
+    echo "Example clusters file content:"
+    echo "myResourceGroup1,myCluster1"
+    echo "myResourceGroup2,myCluster2"
```

Comment:
why not per subscription? or list of subscriptions?  We should have parameter to the  tenantId, subscriptionIdIds, clusterTypes - Aks, Arc k8s.